### PR TITLE
Improve Playback Test Times by Using Better Wait Durations

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR improves how much time running tests in playback uses by performing the following optimizations:

- Change `pollInterval` on `SyncPoller`s and `PollerFlux`s when `TestMode == PLAYBACK`
- Supply custom `RetryPolicy` when testing cases where an operation fails with a retried exception or response type
- Reduced testing payload sizes for generated data

Notable improvements are the following:

| Library | Before | After |
| ------- | ------- | ------ |
| Metrics Advisor | image | image |
| Remote Rendering | image | image |
| Form Recognizer | image | image |
| Communication | image | image |

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientBuilderTest.java [hunk 1]
```diff
@@ -110,6 +110,7 @@ public void timeoutPolicy() {
             ? "Endpoint=http://localhost:8080;Id=0000000000000;Secret=MDAwMDAw"
             : Configuration.getGlobalConfiguration().get(AZURE_APPCONFIG_CONNECTION_STRING);
 
+        // What exactly is this testing?
```
Comment (mssfang) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientBuilderTest.java:113]: It tests if the `TimeoutPolicy" could be working in the client.

**Review by mssfang — APPROVED** [review]
AppConfig, FormRecognizer, TextAnalytics LGTM! Thanks @alzimmermsft

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientBuilderTest.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientBuilderTest.java:113]: Could this be removed for a test in `azure-core`?

**Comment by alzimmermsft** [comment]
@JianpingChen, @ankitarorabit, @minnieliu - could you verify the changes made to Communication libraries.

@samvaity, @anuchandy - could you verify the changes made to Metrics Advisor.

@MalcolmTyrrell - could you verify the changes made to Remote Rendering. Additionally, do you want to be made a [CODEOWNER](https://github.com/Azure/azure-sdk-for-java/blob/master/.github/CODEOWNERS) for `/sdk/remoterendering/` so that you are added as a reviewer to any changes made to that directory?

**Comment by MalcolmTyrrell** [comment]
> @MalcolmTyrrell - could you verify the changes made to Remote Rendering. Additionally, do you want to be made a [CODEOWNER](https://github.com/Azure/azure-sdk-for-java/blob/master/.github/CODEOWNERS) for `/sdk/remoterendering/` so that you are added as a reviewer to any changes made to that directory?

Hi. As far as I can see, the changes to remote rendering look fine. Making me a codeowner makes a lot of sense. Thanks.

**Review by MalcolmTyrrell — APPROVED** [review]

**Pull request merged**